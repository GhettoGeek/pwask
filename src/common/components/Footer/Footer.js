import React from 'react'
import { Grid } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import BottomNavigation from '../BottomNavigation'

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    position: 'fixed',
    bottom: 0,
    left: 0,
    paddingTop: theme.spacing(1),
    paddingBottom: theme.spacing(1),
  },
}))

function Footer() {
  const classes = useStyles()

  return (
    <Grid item xs={12} className={classes.root}>
      <BottomNavigation />
    </Grid>
  )
}

export default Footer
