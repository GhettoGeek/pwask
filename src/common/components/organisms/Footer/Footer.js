import React from 'react'
import PropTypes from 'prop-types'
import { withTranslation } from 'react-i18next'
import { Grid, Typography } from '@material-ui/core'
import { styled } from '../../../utils/style'

const Wrapper = styled('footer')((theme) => ({
  marginTop: theme.spacing(1),
  padding: theme.spacing(1),
}))

const Footer = ({ t }) => (
  <Wrapper>
    <Grid container justify="space-evenly">
      <Typography variant="subtitle1" align="center" color="textSecondary" component="p">
        {t('footer.title')}
      </Typography>
    </Grid>
  </Wrapper>
)

Footer.propTypes = {
  t: PropTypes.func.isRequired,
}

export default withTranslation()(Footer)
