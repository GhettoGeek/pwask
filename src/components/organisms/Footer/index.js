import React from 'react'
import PropTypes from 'prop-types'
import { withTranslation } from 'react-i18next'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { styled } from '../../../utils/style'

const Wrapper = styled('footer')(theme => ({
  marginTop: theme.spacing.s,
  padding: theme.spacing.s,
}))

const Footer = ({ t }) => (
  <Wrapper>
    <Grid container justify="space-evenly">
      <Typography variant="subtitle1" align="center" color="textSecondary" component="p">
        {t('footer.title')}
      </Typography>
    </Grid>
  </Wrapper>
)

Footer.propTypes = {
  t: PropTypes.func.isRequired,
}

export default withTranslation()(Footer)
