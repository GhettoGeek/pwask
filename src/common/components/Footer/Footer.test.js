import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import Footer from './Footer'

describe('Footer', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(
      <MemoryRouter initialEntries={['/']} initialIndex={0}>
        <Footer />
      </MemoryRouter>,
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
})
