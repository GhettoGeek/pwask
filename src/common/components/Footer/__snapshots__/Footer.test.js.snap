// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly the component 1`] = `"<div class=\\"MuiGrid-root makeStyles-root-1 MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiBottomNavigation-root makeStyles-root-105\\"><button class=\\"MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiBottomNavigationAction-wrapper\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z\\"></path></svg><span class=\\"MuiBottomNavigationAction-label Mui-selected\\">common.recents</span></span></button><button class=\\"MuiButtonBase-root MuiBottomNavigationAction-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiBottomNavigationAction-wrapper\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z\\"></path></svg><span class=\\"MuiBottomNavigationAction-label\\">common.favorites</span></span></button><button class=\\"MuiButtonBase-root MuiBottomNavigationAction-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiBottomNavigationAction-wrapper\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z\\"></path></svg><span class=\\"MuiBottomNavigationAction-label\\">common.nearby</span></span></button></div></div>"`;
